adafruit_lsm6ds/__init__.py

```diff
@@ -64,12 +64,18 @@
 from adafruit_register.i2c_bits import RWBits
 from adafruit_register.i2c_bit import RWBit
 
+try:
+    from typing import Tuple, Optional
+    from busio import I2C
```

Comment:
Sorry for the late comment -- but I am confused by this.
If the import of typing fails, won't this then ignore the import of I2C?
Shouldn't the typing import be the last import.

Reply:
@jerryneedell No worries. You are correct. if `typing` fails to import then `I2C` will get ignored.

In this case the only usage of the I2C object that is imported here was for adding type information to some function arguments. CircuitPython will not use that typing information anyway (and will not crash if there are missing imports related to it).

So we use the `import typing` first in order to determine if the library is currently being used on a microcontroller essentially. If `typing` import fails we think it is a microcontroller and therefore we don't need the rest of the imports that are used only for typing either. Putting them after `import typing` and thus skipping them in environments without `typing` will save a little bit of memory I think. If `I2C` did get imported on microcontrollers it would consume a little memory and not actually do anything in this scope.

Reply:
Thank you for the explanation.